src/Stryker.Core/Stryker.Core/Mutators/StringMutator.cs

```diff
@@ -12,7 +14,12 @@ public class StringMutator : MutatorBase<LiteralExpressionSyntax>, IMutator
         public override IEnumerable<Mutation> ApplyMutations(LiteralExpressionSyntax node)
         {
             var kind = node.Kind();
-            if (kind == SyntaxKind.StringLiteralExpression && !(node.Parent is ConstantPatternSyntax))
+            // Get objectCreationSyntax to check if it contains a regex type.
+            var root = node.Parent?.Parent?.Parent;
+
+            if (!(root is ObjectCreationExpressionSyntax parsedRoot && IsRegexType(parsedRoot))
+                && kind == SyntaxKind.StringLiteralExpression
+                && !(node.Parent is ConstantPatternSyntax))
```

Comment:
Could you perhaps also extract these two conditions to private helper functions with clear names similar to IsRegexType so the condition becomes easier to read?